x/blob/types/payforblob.go

```diff
@@ -49,7 +49,7 @@ const (
 var _ legacytx.LegacyMsg = &MsgPayForBlobs{}
 
 func NewMsgPayForBlobs(signer string, blobs ...*Blob) (*MsgPayForBlobs, error) {
-	err := ValidateBlobs(blobs...)
+	err := ValidateBlobs(appconsts.LatestVersion, blobs...)
```

Comment:
On second thought I think this may be dangerous for similar reasons mentioned in https://github.com/celestiaorg/celestia-app/pull/2203#discussion_r1283420399

This assumes the latest celestia-app version so the behavior varies based on when a node upgrades.